networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/provider_nsx_policy.py

```diff
@@ -618,6 +617,33 @@ def _wait_to_realize(self, resource_type, os_id):
         exporter.REALIZED.labels(resource_type, status).inc()
         raise Exception("{} ID: {} did not get realized for {}s", resource_type, os_id, until * pause)
 
+    def _realize_bulk(self, resource_type: str, data, nsxt_res_type, dsl):
+        report = "Bulk Operation: Resource: {} with ID: {} is going to be %s.".format(resource_type, lambda: data[0]["id"])
+
+        LOG.info(report, "created")
+        path = "{}".format(API.INFRA)
+        
+        res = self.client.patch(path=path, data=data)
```

Comment:
We need a lock before the execution of the actual update/create operation. The challenge here is not to lock on the entire "SG_MEMBERS" type but for each group ID individually "member-{}".format(os_id). A semaphore implementation would be required.